src/lua/LuaInput.cpp

```diff
@@ -2,21 +2,245 @@
 // Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
 
 #include "LuaInput.h"
+
+#include "ConnectionTicket.h"
 #include "GameConfig.h"
 #include "Input.h"
 #include "InputBindings.h"
-#include "Lang.h"
 #include "LuaMetaType.h"
 #include "LuaObject.h"
 #include "LuaUtils.h"
 #include "LuaWrappable.h"
 #include "Pi.h"
 
-#include "SDL_keyboard.h"
+#include "core/StringUtils.h"
+
+#include <SDL_keyboard.h>
+#include <SDL_keycode.h>
 #include <sstream>
 
 using namespace InputBindings;
 
+static std::vector<std::pair<const char *, int>> s_input_keys = {
+	// Symbol keys
+
+	{ "exclaim", SDLK_EXCLAIM },
```

Comment:
Q: Why have the LUA symbols lowercase instead of uppercase? 

It makes searching for constants harder (I tend to use case-sensitive searches when exploring the codebase to reduce the number of false positives)

Also: aren't strings significantly less performant than integer indeces (No idea if LUA has a concept of enums)?

Reply:
> Why have the LUA symbols lowercase instead of uppercase?

1. For compatibility with existing code - this matches the existing `ui.keys.xxxx` defines and directly replaces them.
2. I happen to like the look of `Input.keys.delete` more than I like `Input.keys.DELETE`, and it's more conformant to the closest thing we have to a [style guide](https://forum.pioneerspacesim.net/viewtopic.php?p=6293#p6293).

> Also: aren't strings significantly less performant than integer indeces (No idea if LUA has a concept of enums)?

These are named aliases to integer values. They will typically be looked up once at the start of the program when registering key bindings, so performance isn't really an issue. This is effectively the Lua concept of an enum.

Note that in Lua, all strings are immutable and interned, so comparing strings is a direct pointer comparison and equal strings have a shared "pointer identity" used to look up the string in a hashmap (i.e. as a table key).